### ISSUE TITLE
iOS and Android configuration?

### ISSUE THREAD

**Issue by somefakeuser** [arleyandrada/PushClient#21]
Hello. I am sorry if my english is not right, i am from spain...
Maybe you could send me an example of ios and android configuration?? In the example into the module i think is only for android...
Thanks a lot.

**Comment by arleyandrada** [comment]
Hi,

The example from my module documentation is the same for both iOS and Android.
​
Do you have any specific questions or difficulty that I can help?

Regards,

Arley

**Comment by somefakeuser** [comment]
Hi again... I put the code in my app.js and there isnt no way to recibe the eventSuccess and i can not get the device token (registrationId). Nothing happen...noone event fire...
Am i doing something wrong??
I installed and AdHoc version in my iphone and i think that all my certificates are right installed.
What can i do??

Thanks very much!!

**Comment by arleyandrada** [comment]
Hi,

You may have noticed that the module does not work on the emulator, but you should have found some kind of message in the Titanium console.

To get a successfully device registration, both on iOS as on Android, you must register your application on Apple or Google and set it up to use push notifications.

Unfortunately my module is not yet compatible with Titanium Studio 3.4 or Titanium SDK 3.4, but work fine with iOS 8 if build with previous Titanium version.

Some useful links: 

https://developer.apple.com/library/mac/documentation/NetworkingInternet/Conceptual/RemoteNotificationsPG/Introduction.html

https://parse.com/tutorials/ios-push-notifications

Please let me know if you get it to work or have any problems.

Regards,

Arley

**Comment by somefakeuser** [comment]
Hi Arley, thanks for your answer.
I know that the module does not work  on emulator and i have registered with push notification my app in Apple and google. 
The version of my Titanium is 3.3.0.GA. 
When i put the code and run the app in my iphone i received the alert that i have to let the iphone receive push notification but no more alerts were shown of the events (if error or success) so i can not get the device token for sending push notif. 

About the document you send me, do i need to be registered in parse for sending push notif. and registering the device?

thanks!!

**Comment by arleyandrada** [comment]
Hi,​

I don't know what is wrong in your app.
Maybe you can share it with me (entire project or minimum version of it).

No, you don't need to use Parse.com as your push server.
You can use any push notification server.

Regards,

Arley

**Comment by arleyandrada** [comment]
Hi,​

Did you put the "issue #17030 workaround" below in your app source code?

```
//Workaround to Titanium Mobile issue #17030
//https://jira.appcelerator.org/browse/TIMOB-17030
if (!OS_ANDROID) {
    var issue17030 = Ti.Network.registerForPushNotifications;
}
```

You need to put it in any place of your source code.

Please let me know if it works to you.

Regards,

Arley

**Comment by somefakeuser** [comment]
Hi Arley thanks so much for your time. This morning i discovered something strange in Titanium...I changed everything and nothing worked, finally i did a new proyect and works...after that i did a new app.js file in my proyect and copied the code from the old app.js and...works!!! It has the same code and i dont know why now works. I tested with the original function of titanium not with your module but i think with your module have to work too. 
Now i can not test it but when i can i tell you what happen, if it works or not. 
Regards.
